src/ashley/machina_extensions/forum_conversation/forms.py

```diff
@@ -0,0 +1,55 @@
+"""
+    Forum conversation forms
+    ========================
+
+    This module defines forms provided by the ``forum_conversation`` application.
+
+"""
+import logging
+
+from django.urls import reverse
+from machina.apps.forum_conversation.forms import PostForm as MachinaPostForm
+from machina.core.db.models import get_model
+from machina.core.loading import get_class
+
+get_forum_member_display_name = get_class(
+    "forum_member.shortcuts", "get_forum_member_display_name"
+)
+User = get_model("ashley", "User")
+logger = logging.getLogger(__name__)
+
+
+class PostForm(MachinaPostForm):
+    """ Overload Machina PostForm to send extra variables to the widget editor """
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        # collect active users for this topic
+        active_post_users = (
+            self.topic.posts.filter(approved=True, poster__is_active=True)
+            .exclude(poster=self.user)
+            .values("poster", "username")
+            .order_by("poster")
+            .distinct()
+        )
+
+        list_active_users = [
+            {
+                "name": get_forum_member_display_name(
+                    User.objects.get(pk=user["poster"])
+                )
+                if User.objects.get(pk=user["poster"])
+                else user["username"],
+                "link": reverse("forum_member:profile", kwargs={"pk": user["poster"]}),
+            }
+            for user in list(active_post_users)
+        ]
+        self.fields["content"].widget.attrs["mention_users"] = sorted(
+            list_active_users, key=lambda i: i["name"]
+        )
```

Comment:
- I noticed that you are ordering twice, so ordering the first query is useless
- You are filtering the first query on active users so you are missing the anonymous users that only keyed in their username without being authenticated (https://github.com/ellmetha/django-machina/blob/master/machina/apps/forum_conversation/abstract_models.py#L245). 
- I think you should probably handle anonymous users and authenticated users in separate queries. For authenticated users, a query on the User model will be more natural. For anonymous users, the "link" property will by null because they don't have a profile right?

Reply:
For anonymous users:
```
anonymous_posters = (
    self.topic.posts.filter(approved=True, poster__is_null=True, username__is_null=False)
    .values("username", flat=True)
)
```

For authenticated users:
```
authenticated_users = (
    User.objects.filter(is_active=True).filter(posts__topic=self.topic, posts__approved=True)
    .exclude(poster=self.user)
)
```
    .

Reply:
Sorry Sam I didn't read and came myself exactly at the same time with a fixup I had trouble yesterday with this query and realised this morning I could avoid caring about anonymous users by requesting from users directly. Could you check my fixup to see if I already answered your comment

Reply:
:astonished: 
Ok so you are dropping the possibility to mention anonymous users. Maybe it will be confusing for users that some usernames can be mentioned and others not? I was thinking to allowing mentions and display them without any link. maybe it does not make sense :man_shrugging:?

Reply:
At the moment we don't have this use case, in Django machina it's possible but not in Ashley. Each user are authenticated through LTI. It would be weird to deal with a case that we don't have and do extra work. It makes it more simple and realistic to ignore this use case. If we were allowing anonymous users to post messages in the future which is not certain, we still would be able to add them to mention them.

Reply:
True! :clap: